src/itdelatrisu/opsu/beatmap/BeatmapParser.java

```diff
@@ -589,6 +589,8 @@ else if ((type & HitObject.TYPE_SLIDER) > 0)
 		// parse hit objects now?
 		if (parseObjects)
 			parseHitObjects(beatmap);
+		
+		beatmap.md5Hash = Utils.getMD5(file);
```

Comment:
There should be a way to calculate the MD5 without reading the file twice -- we should definitely do that instead, since this will probably hurt loading times.